spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -1240,13 +1240,27 @@ def request_hover(self, line, col, offset, show_hint=True, clicked=True):
     @handles(LSPRequestTypes.DOCUMENT_HOVER)
     def handle_hover_response(self, contents):
         """Handle hover response."""
+        try:
+            from unittest.mock import Mock
+        except ImportError:
+            from mock import Mock  # Python 2
+
+        # On some tests this is returning a Mock
+        if isinstance(contents, Mock):
+            return
+

```

Comment:
Please move all these new lines inside an `if running_under_pytest():` because it only applies to our tests.